apps/api/src/document/dto/create-document.dto.ts

```diff
@@ -0,0 +1,58 @@
+
+import { DocumentType, Prisma } from '@prisma/client'
+import { ApiProperty } from '@nestjs/swagger'
+import { IsNotEmpty, IsObject, IsOptional, IsString, ValidateNested } from 'class-validator';
+import { Expose, Type } from 'class-transformer';
+import { CreatePersonDto } from '../../domain/generated/person/dto';
+
+@Expose()
+export class CreateDocumentDto {
+    @Expose()
+    @ApiProperty({ enum: DocumentType })
+    type: DocumentType;
+
+    @Expose()
+    @ApiProperty()
+    @IsString()
+    name: string;
+
+    @Expose()
+    @ApiProperty()
+    @IsString()
+    filename: string;
+
+    @Expose()
+    @ApiProperty()
+    @IsString()
+    @IsOptional()
+    filetype: string;
+
+    @Expose()
+    @ApiProperty()
+    @IsString()
+    @IsOptional()
+    description: string;
+
+    @Expose()
+    @ApiProperty()
+    @IsString()
+    sourceUrl: string;
+
+
+    public toEntity(): Prisma.DocumentCreateInput {
+        return {
+            type: this.type,
+            name: this.name,
+            filename: this.filename,
+            filetype: this.filetype,
+            description: this.description,
+            sourceUrl: this.sourceUrl,
+            owner: {
+                //hardcoded for now
+                connect: {
+                    id: '15661b63-e72f-4c30-8d7d-a75e8d1b69f4'
```

Comment:
Please remove the hardcoded part so we can merge this PR once it's done

Reply:
@kachar Is it okay to create the owner by passing down the keycloak token like this
From Controller:
`return this.documentService.create(createDocumentDto, user);`
To Service:
`async create(createDocumentDto: CreateDocumentDto, user: KeycloakTokenParsed) {
    return await this.prisma.document.create({ data: createDocumentDto.toEntity(user) })
  }`
To createDocument.toEntity() method in order to connect an existing user or create one according to the data coming from the token?
` owner: {
                connectOrCreate: {
                    where: {
                        email: user.email
                    },
                    create: {
                        firstName: user.given_name,
                        lastName: user.family_name,
                        email: user.email
                    }
                }
            },`

Reply:
Yep, this way of handling the owner seems to be good at this point as we don't have a generic approach yet.

Thanks for updating the codebase